wfl/Database/MongoDBManager.cs

```diff
@@ -8,55 +8,31 @@
 
 namespace Symphono.Wfl.Database
 {
-    public class MongoDBManager: IDBManager
+    public class MongoDBManager<T>: IDBManager<T> where T: IEntity
     {
         private static string connectionString;
         public static MongoClient client;
         public static IMongoDatabase db;
+        private string collectionName { get; }
```

Comment:
So I think there are a couple definitions for "read only"...having no ` { set; } ` means you can't set the value using code, however, I'm pretty sure you can still set the value using reflection. By using the `readonly` keyword, the field basically becomes immutable.